### PR TITLE
Getting rid of a static ssh password and renaming zen to eve

### PR DESCRIPTION
**Pull request by rvs**
Signed-off-by: Roman Shaposhnik <rvs@zededa.com>

### CODE REVIEW

File: pkg/pillar/rootfs/init.sh [hunk 1]
```diff
@@ -11,8 +11,14 @@ for i in `cd /sys/class/net ; echo eth*` ; do
...
-# For convenice's sake we're putting SSH inisde of a root container 
-/usr/sbin/sshd
+# constructing /run/authorized_keys from a static one
+# At some point we may also consider extracting key material
+# from x509 certificates...
+#   bash -c 'ssh-keygen -f <(openssl x509 -in /config/onboard.cert.pem -pubkey -noout) -i -mPKCS8' >> /run/authorized_keys
+# ...or using CA signed ssh keys
+#   TrustedUserCAKeys /path/to/server_ca.pub
+cp /config/authorized_keys /run
+chmod 700 /run/authorized_keys
```
Comment (deitch) [inline pkg/pillar/rootfs/init.sh]: I am pretty sure `authorized_keys` should be `0600` and not `0700`. That having been said, there are various reports of it needing to be `0644` (which is a bit strange). I just checked three different distros for their default:

* CentOS: `0600`
* Ubuntu: `0600`
* CoreOS: `0644`

We definitely should use `0600` or `0644` but not `0700`.

**Comment by deitch** [comment]
Comment from @eriknordmark :

> My idea a week ago or so was to pass an authorized key using a config item. Whether we do by redefining debug.enable.ssh from a boolean to the string with a key requires some care to not mess up the devices we have in the lab. I'd suggest defining a new config item and then later retiring the boolean.
>
> For us and an EVE developer, is it easier to use the CA key approach vs. just specifying the key on my laptop? I think for an EVE developer the CA approach seems overkill.
>
> If Zededa engineers need to debug devices in production then a single authorized key should suffice as well.
>For our lab setup with alpha/hummingbird a CA key might make sense, but that requires managing our ssh keys that way. And I'd really want 95%+ of our work there to use the logs and TBD APIs to extract state out of the device, and depend a lot less on ssh.

**Comment by deitch** [comment]
Comment from @rvs:

>If it is just the opaque config item -- it is exactly the same as what I proposed in this thread.
>
> At this point, I feel that the choice is between picking CA vs. authorized_keys approaches. Let me come up with a bit more background on either of these before we commit to a final solution.

**Comment by deitch** [comment]
Comment from @rvs:

> while moving it I realized that we actually lost an ability to ssh directly into the zedctr/pillar container with this patch. Do you think yo can help coming up with some clever trick for setting a shell for root so that it immediately does

```ctr --namespace services.linuxkit t exec -t --exec-id XXX  pillar sh```

**Comment by deitch** [comment]
@rvs:

> At this point, I feel that the choice is between picking CA vs. authorized_keys approaches

I think we all are on the same page of the `authorized_keys` approach at this point, and your PR reflects it. Is that correct?

It is just a question as to whether `debug.enable.ssh` converts from boolean to string (wherein the string contains the public key to put in `authorized_keys`) or it is a separate parameter. I strongly prefer a single parameter to reflect _both_ its enablement _and_ its content, but understand what @eriknordmark said about not breaking existing things.

Erik's proposal that we create a new key, perhaps `debug.ssh.keys` as a string, and immediately deprecate and then eventually retire `debug.enable.ssh`, appears to be the most logical path forward.

**Comment by deitch** [comment]
@rvs wrote:

> while moving it I realized that we actually lost an ability to ssh directly into the zedctr/pillar container with this patch. Do you think yo can help coming up with some clever trick for setting a shell for root so that it immediately does

```ctr --namespace services.linuxkit t exec -t --exec-id XXX pillar sh```

I think my preference would be to do this not at the shell level but at the ssh config level, probably in  `authorized_keys`; let me think about it. 

However, an impact of this is that the remote user would _only_ be in the pillar container, and _not_ in the root OS. They would have no way to get out (well, no sane way, and I am not going to try and prove Jess wrong with https://contained.af ; she probably is right!). 

Let me propose that we do this iteratively:

1. Use this PR as is
2. Add another short PR that creates an alias for the above comment, so you can just type `pillar` once you are ssh'ed in and get where you need to be.
3. Add a final PR that automatically runs the command (via sh or via ssh config) that effectively executes it, if this is what we want. 

Make sense?

**Comment by eriknordmark** [comment]
> It is just a question as to whether `debug.enable.ssh` converts from boolean to string (wherein the string contains the public key to put in `authorized_keys`) or it is a separate parameter. I strongly prefer a single parameter to reflect _both_ its enablement _and_ its content, but understand what @eriknordmark said about not breaking existing things.

We can have the parser in zedagent look for "true", "false" and an ascii blob with a key. That would provide backwards compatibility.

But I don't think we need to do that in this PR; just add the file in /config

File: pkg/pillar/rootfs/init.sh (hunk 1 above)
Comment (rvs) [reply pkg/pillar/rootfs/init.sh]: Great point @deitch will change this to 0600

**Comment by rvs** [comment]
Committing this one as an MVP. Next steps: figure out whether it would be easier to have a CA that can sign everybody's keys OR follow-up on @nordmark idea of having config knob being able to add custom keys at runtime

**Pull request merged**